jitsi-utils-kotlin/pom.xml

```diff
@@ -1,148 +1,136 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-    <modelVersion>4.0.0</modelVersion>
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
 
-    <parent>
-        <groupId>org.jitsi</groupId>
-        <artifactId>jitsi-utils-parent</artifactId>
-        <version>1.0-SNAPSHOT</version>
-    </parent>
+  <parent>
+    <groupId>org.jitsi</groupId>
+    <artifactId>jitsi-utils-parent</artifactId>
+    <version>1.1-SNAPSHOT</version>
+  </parent>
 
-    <artifactId>jitsi-utils-kotlin</artifactId>
+  <artifactId>jitsi-utils-kotlin</artifactId>
+  <!--
+  bundle instead of jar for OSGi capabilities. Note that this merges the
+  classes of jitsi-utils (the java-only package) into this bundle, thus the
+  compile-only scope below.
+  -->
+  <packaging>bundle</packaging>
 
-    <properties>
-        <kotlin.version>1.3.71</kotlin.version>
-    </properties>
+  <properties>
+    <kotlin.version>1.3.71</kotlin.version>
+  </properties>
 
-    <dependencies>
-        <dependency>
-            <groupId>org.jitsi</groupId>
-            <artifactId>jitsi-utils</artifactId>
-            <version>${project.version}</version>
-        </dependency>
-        <dependency>
-            <groupId>org.jetbrains.kotlin</groupId>
-            <artifactId>kotlin-stdlib-jdk8</artifactId>
-            <version>${kotlin.version}</version>
-        </dependency>
-        <dependency>
-            <groupId>org.jetbrains.kotlin</groupId>
-            <artifactId>kotlin-reflect</artifactId>
-            <version>${kotlin.version}</version>
-        </dependency>
+  <dependencies>
+    <dependency>
+      <groupId>org.jitsi</groupId>
+      <artifactId>jitsi-utils</artifactId>
+      <version>${project.version}</version>
+      <!-- See note above about the packaging type -->
+      <scope>compile</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.github.spotbugs</groupId>
+      <artifactId>spotbugs-annotations</artifactId>
+      <optional>true</optional>
+    </dependency>
+    <dependency>
+      <groupId>org.jetbrains</groupId>
+      <artifactId>annotations</artifactId>
+      <optional>true</optional>
+    </dependency>
+    <dependency>
+      <groupId>org.jetbrains.kotlin</groupId>
+      <artifactId>kotlin-osgi-bundle</artifactId>
```

Comment:
So I understand: why does this obsolesce the requirement for kotlin-reflect?

Reply:
> To enable Kotlin OSGi support you need to include kotlin-osgi-bundle instead of regular Kotlin libraries. It is recommended to remove kotlin-runtime, kotlin-stdlib and kotlin-reflect dependencies as kotlin-osgi-bundle already contains all of them.

https://kotlinlang.org/docs/reference/kotlin-osgi.html

Reply:
Hm, also from there:

> Most regular Kotlin dependencies are not OSGi-ready, so you shouldn't use them and should remove them from your project.

This is the sort of thing that doesn't make me thrilled about changing this to be OSGi.

Reply:
The non-readyness is true for many, if not most, regular Java libraries as well. It doesn't mean it's not possible to use them, but it will require more effort in applications running in an OSGi environment, i.e. Jitsi Desktop. All other applications won't bother.

Reply:
So, another dumb question, is the 'bundle-able' requirement for all dependencies?  That is, must all the libs `jitsi-utils-kotlin` depend be bundle-friendly?  Or is it just the direct dependency itself (jitsi-utils-kotlin)?

Reply:
It's the entire chain. In jitsi-rtp e.g. the `unsigned` library is such a non-osgi dependency that would hinder things. But since Kotlin itself now has unsigned types (although experimental, but other experimental features are also enabled there)...

Reply:
So, this would require everything `jitsi-utils-kotlin` depends on to be bundle-friendly?

Reply:
Yes, this is what JetBrains means with `Most regular Kotlin dependencies ...`

Reply:
> It's the entire chain. In jitsi-rtp e.g. the `unsigned` library is such a non-osgi dependency that would hinder things. But since Kotlin itself now has unsigned types (although experimental, but other experimental features are also enabled there)...

But didn't you update libjitsi to use jitsi-srtp?  Is it just that you haven't changed jitsi-desktop to use that version of libjitsi yet?

Reply:
Check your e-mail for a Meet-link please :-)